client/allocrunner/cni/bridge.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package cni
```

Comment:
It feels a little weird having a separate package called `cni` here without it including much of any of the CNI configuration. There's no code isolation happening because everything is exported (has to be because of serialization.) All CNI setup including all the configuration tests are still in the `client/allocrunner` package. It seems hard to disentangle the "bridge configurator" code from CNI as well.

Reply:
the initial motivation was to namespace the structs -- I found myself naming them like `FirewallCNIPlugin` just because they were in the `allocrunner` package, and I thought if I made a small new package, then they could be named more simply and still make sense.  I also just like small, focused packages in general.

but were I to spend more time on this, I probably would move more of the CNI stuff into here, but I figure this is an incremental improvement to be perhaps built on further, later.

regarding the "networking" package comment, a decent chunk of our networking config, especially on linux, is CNI, but I might still like to distinguish the things that are specifically satisfying CNI spec from other more-general networking concerns.  i.e. if I really went ham refactoring this area of allocrunner, I might make a "networking" package, then still put a (relatively smaller) "cni" package within there.

but as you might imagine, I am trying not to boil the ocean! :sweat_smile: 

all that said, I could move the new golden-file tests into here, since they are a direct output of this package.  I left them where they were because that was the hook into this refactor, but I did create them here in the first place...

Reply:
I'll also admit to being a little concerned about backports, since I've been intending to go back to 1.6.x for future maintainability, but e.g. consul-cni is 1.8.x only.  backports won't be tooooo painful anyway, or I could forego them entirely, but at any rate, it was on my mind when deciding what all to move into this new package.

Reply:
> but as you might imagine, I am trying not to boil the ocean!

Totally reasonable :+1: